src/main/java/org/mskcc/cbio/oncokb/service/MailService.java

```diff
@@ -34,75 +43,153 @@
-            message.setText(content, isHtml);
-            javaMailSender.send(mimeMessage);
-            log.info("Sent email to User '{}'", to);
-        }  catch (MailException | MessagingException e) {
-            log.warn("Email could not be sent to user '{}'", to, e);
-        }
+        MimeMessageHelper message = new MimeMessageHelper(mimeMessage, isMultipart, StandardCharsets.UTF_8.name());
+        message.setTo(to);
+        message.setFrom(from);
+        message.setSubject(subject);
+        message.setText(content, isHtml);
+        javaMailSender.send(mimeMessage);
+    }
+
+    private void addUserMailsRecord(UserDTO userDTO, MailType mailType, String sentFrom, String sentBy) {
+        UserMailsDTO userMailsDTO = new UserMailsDTO();
+        userMailsDTO.setMailType(mailType);
+        userMailsDTO.setSentDate(Instant.now());
+        userMailsDTO.setSentFrom(sentFrom);
+        userMailsDTO.setSentBy(sentBy);
+        userMailsDTO.setUserId(userDTO.getId());
+        userMailsService.save(userMailsDTO);
+    }
+
+    @Async
+    public void sendEmailFromTemplate(UserDTO user, MailType mailType) {
+        sendEmailFromTemplate(user, mailType, jHipsterProperties.getMail().getFrom(), jHipsterProperties.getMail().getFrom());
     }
 
     @Async
-    public void sendEmailFromTemplate(User user, String templateName, String titleKey) {
+    public void sendEmailFromTemplate(UserDTO user, MailType mailType, String from, String by) {
         Locale locale = Locale.forLanguageTag(user.getLangKey());
         Context context = new Context(locale);
         context.setVariable(USER, user);
         context.setVariable(BASE_URL, jHipsterProperties.getMail().getBaseUrl());
-        String content = templateEngine.process(templateName, context);
-        String subject = messageSource.getMessage(titleKey, null, locale);
-        sendEmail(user.getEmail(), subject, content, false, true);
+        context.setVariable(LICENSE, user.getLicenseType().getName());
+        String content = templateEngine.process("mail/" + mailType.getTemplateName(), context);
+        String subject = messageSource.getMessage(getTitleKeyByMailType(mailType).orElseThrow(() -> new UnknownMailTypeException()), new Object[] {user.getLicenseType().getName()}, locale);
+        try {
+            if (from == null) {
+                from = jHipsterProperties.getMail().getFrom();
+            }
+            if (by == null) {
+                by = from;
+            }
+            sendEmail(from, user.getEmail(), subject, content, false, true);
+            addUserMailsRecord(user, mailType, from, by);
+            log.info("Sent email to User '{}'", user.getEmail());
+        } catch (MailException | MessagingException e) {
+            log.warn("Email could not be sent to user '{}'", user.getEmail(), e);
+        }
     }
 
     @Async
-    public void sendActivationEmail(User user) {
+    public void sendActivationEmail(UserDTO user) {
```

Comment:
Add `activation_key` in UserDTO.java since it will be used in the email template activationEmail.html later.

Reply:
Added.